pandas/core/internals.py

```diff
@@ -1070,7 +1070,16 @@ def eval(self, func, other, raise_on_error=True, try_cast=False, mgr=None):
                                      "values [%s]" % (values.T.shape,
                                                       other.shape))
 
-        transf = (lambda x: x.T) if is_transposed else (lambda x: x)
+        def transf(x):
+            if is_transposed:
+                x = x.T
+
+            # fix issue #12564: CategoricalBlock is 1-dim only, while category
+            # datarame can be more.
+            if getattr(self, 'is_categorical', None) and x.ndim < self.ndim:
```

Comment:
1. `x.ndim < self.ndim`        
    if `x` is a normal object, `_block_shape` will increase `x.ndim` so that  `ValueError: Wrong number of dimensions` raises. Thus I check `x.ndim < self.ndim`.

2. `getattr(self, 'is_categorical', None)`          
     Because we expect that `x.ndim < self.ndim` should only occur in `NonConsolidatableMixIn` class, so if it is not the case, I believe that it might be a bug.